paradox-nakadi-consumer-sqs-error-handler/src/test/java/de/zalando/paradox/nakadi/consumer/sqserrorhandler/SQSFailedEventSourceTest.java

```diff
@@ -0,0 +1,243 @@
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.apache.commons.lang3.RandomUtils.nextLong;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import static de.zalando.paradox.nakadi.consumer.sqserrorhandler.SQSFailedEventSource.EVENT_SOURCE_NAME;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.amazonaws.http.SdkHttpMetadata;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.DeleteMessageResult;
+import com.amazonaws.services.sqs.model.GetQueueAttributesRequest;
+import com.amazonaws.services.sqs.model.GetQueueAttributesResult;
+import com.amazonaws.services.sqs.model.GetQueueUrlResult;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.QueueAttributeName;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.zalando.paradox.nakadi.consumer.core.domain.EventType;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+public class SQSFailedEventSourceTest {
+
+    private SQSFailedEventSource sqsFailedEventSource;
+
+    private ObjectMapper objectMapper;
+
+    private SQSConfig sqsConfig;
+
+    private AmazonSQS amazonSQS;
+
+    @Before
+    public void setUp() {
+        amazonSQS = mock(AmazonSQS.class);
+        sqsConfig = mock(SQSConfig.class);
+        objectMapper = new ObjectMapper();
+
+        final String queueName = randomAlphabetic(10);
+        when(sqsConfig.getQueueName()).thenReturn(queueName);
+
+        final GetQueueUrlResult getQueueUrlResult = new GetQueueUrlResult();
+        final SdkHttpMetadata responseMetadata = mock(SdkHttpMetadata.class);
+        when(responseMetadata.getHttpStatusCode()).thenReturn(200);
+        getQueueUrlResult.setSdkHttpMetadata(responseMetadata);
+        getQueueUrlResult.setQueueUrl(randomAlphabetic(10));
+
+        when(amazonSQS.getQueueUrl(queueName)).thenReturn(getQueueUrlResult);
+
+        sqsFailedEventSource = new SQSFailedEventSource(sqsConfig, amazonSQS, objectMapper);
+    }
+
+    @Test
+    public void testShouldReturnSQSEventSourceName() {
+        assertThat(sqsFailedEventSource.getEventSourceName()).isEqualTo(EVENT_SOURCE_NAME);
+        verify(amazonSQS).getQueueUrl(anyString());
+    }
+
+    @Test
+    public void testShouldFailWhenQueueUrlIsNotRetrieved() {
+
+        amazonSQS = mock(AmazonSQS.class);
```

Comment:
Initilization happens in `setUp` already. It's unclear why objectMapper is a mock here (in `setUp` it is a real instance)